### ISSUE TITLE
Test run seriously slowing down when renaming contracts folder

### ISSUE THREAD

**Issue by alexykot** [pipermerriam/populus#368]
* Populus Version: 1.11.0
* OS: linux/ubuntu

### What was wrong?
consider folder structure:
```
contracts/
tests/
project.json
README.md
```
`tests` folder has 52 items to run, running them all with command `py.test -s --disable-pytest-warnings --populus-project=./ ./` takes roughly ~85 seconds, maxing out load on one CPU core on my old laptop.

Now consider this folder structure:
```
ws_contracts/
tests/
project.json
README.md
```
Same files inside, `project.json` amended to reference the right folder. 
Running same tests with the same command takes 4-5 times longer, also maxing out load on one CPU core on my old laptop. No apparent reason why.

Expected: renaming the contracts folder should not affect the test running time.

**Comment by veox** [comment]
I've tried this with the `populus init`-made `Greeter` project, and noticed that the `ws_contracts` test run, executed as plain `py.test`, produces several times more deprecation warnings. The "extra" ones seem to be mostly:

``` python
/<path snipped>/lib/python3.6/site-packages/populus/project.py:235: DeprecationWarning: project.contracts_source_dir has been replaced by the plural, project.contracts_source_dirs which is an iterable of all source directories populus will search for contracts.  Please update your code accordingly as this API will be removed in a future release
    "project.contracts_source_dir has been replaced by the plural, "
```

(The trunctation of the second line appears in both `contracts` and `ws_contracts` run.)

Consider comparing the generated warnings (that you currently discard). These may point to the root of the issue.

**Comment by veox** [comment]
BTW, which Python version are you running?

**Comment by alexykot** [comment]
Python 3.5.2 (default, Sep 14 2017, 22:51:06). Added to the issue description.

**Comment by alexykot** [comment]
Yes, I discard general warnings because they point to all issues in all dependencies that is not really helpful, and as you see there is a lot of them. I'll check that though.
